src/main/java/org/opentripplanner/updater/bike_rental/GbfsBikeRentalDataSource.java

```diff
@@ -161,4 +172,44 @@ public BikeRentalStation makeStation(JsonNode stationNode) {
             return brstation;
         }
     }
+
+    /**
+     * This is ugly because unlike the others, it does not return a list of rental station information.
+     * It assembles some global opening hours information about all the stations in the system.
+     * For the time being I'm making this extend GenericJsonBikeRentalDataSource so it reuses
+     * the HTTP/local file fetching code, JSON decoding, etc. but those methods are closely coupled to the
+     * decoding of individual station representations. Those methods should eventually be pulled out and
+     * made into more reuseable static methods that are not so closely tied to the decoding of stations.
+     *
+     * This updater is also a little strange in that instead of returning a list of new objects, it
+     * updates its own internal field that contains the operating hours, and provides methods to check
+     * the contents of that list.
+     */
+    class GbfsSystemHoursDataSource extends GenericJsonBikeRentalDataSource {
+
```

Comment:
Another way to do it is to make 2 abstract classes: `GenericJsonBikeRentalDataSource` with an abstract `processNodes()` method and `IndividualStationJsonBikeRentalDataSource`.

Reply:
I have added your comment to the Javadoc. This certainly needs to be refactored someday but I am just adding on to / working within a pre-existing GBFS structure that was itself a bit awkward to fit into the existing bike rental source framework.